test-suite/output/Implicit.out

```diff
@@ -5,7 +5,7 @@ ex_intro (P:=fun _ : nat => True) (x:=0) I
 d2 = fun x : nat => d1 (y:=x)
      : forall x x0 : nat, x0 = x -> x0 = x
 
-Arguments d2 [x x0]%nat_scope
+Arguments d2 [x x]%nat_scope
```

Comment:
Would it make sense here to go one step further and to consider that the second argument `x` of `d2` is unavailable by name (forcing a `_` in `Arguments`), on the basis that the user deliberatedly chose `x` as first argument of `d2`?

Or to warn at the time of defining `d2`?

Reply:
Something like that does indeed sound good, yes.

The code change required is more than just doing the deduplication in `compute_implicits_names_gen`; anonymous implicit arguments there adopt the name from the manual implicit information somewhere in `prepare_implicits` (I think), and I haven't put in the time to re-understand the logic of `prepare_implicits` in the light of anonymous implicits.

Warning might be tricky, since for example a `Proper (R1 ==> R2 ==> R3)` instance gets unfolded to find argument names of `x y _ x y _`, which might warn without special handling to silence warnings after unfolding.

I might also advocate for keeping only the *last* argument of the same name, since that notion is stable under discharging sections, but that is unintuitive in it's own way on the `Proper` example.

Reply:
I think it's ok to tolerate duplicated names, like OCaml does for labels.